model/src/main/kotlin/utils/FileArchiver.kt

```diff
@@ -68,15 +68,20 @@ class FileArchiver(
      * Return whether an archive corresponding to [provenance] exists.
      */
     fun hasArchive(provenance: Provenance): Boolean {
-        val archivePath = getArchivePath(provenance)
+        if (provenance.sourceArtifact == null && provenance.vcsInfo == null) return false
```

Comment:
I'm not against this per se, but would like to understand why we do not put this check into the constructor of the provenance? Any thoughts?

Reply:
The check does actually exists. So let's drop these changes here.

Reply:
The existing check is different:

https://github.com/oss-review-toolkit/ort/blob/f3d2aa5d957d995593fa0d7a793b3b49670759e5/model/src/main/kotlin/Provenance.kt#L65-L67

Note the `&&` instead of `||`. Because we actually *do* allow to capture "empty" `Provenance` so far (which we probably shouldn't, but that's a topic for itself), see e.g.

https://github.com/oss-review-toolkit/ort/blob/f3d2aa5d957d995593fa0d7a793b3b49670759e5/scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml#L220-L221

But here, we need to avoid an empty `Provenance`.

Reply:
Oh, that existing check in `Provenance.kt` IMO is quite a bit counter intuitive. In fact it seems wrong.
Anyhow, let's merge this to get rid of the NPE quickly.